### PR TITLE
STY: "{foo!r}" -> "{repr(foo)}" #batch-3

### PR DESCRIPTION
**Pull request by ShaharNaveh**
- [x] ref #29886
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/nanops.py [hunk 1]
```diff
@@ -60,8 +60,10 @@ def __call__(self, f):
         def _f(*args, **kwargs):
             obj_iter = itertools.chain(args, kwargs.values())
             if any(self.check(obj) for obj in obj_iter):
-                msg = "reduction operation {name!r} not allowed for this dtype"
-                raise TypeError(msg.format(name=f.__name__.replace("nan", "")))
+                f_name = f.__name__.replace("nan", "")
```
Comment (MomIsBestFriend) [inline pandas/core/nanops.py]: Can anyone think of a better name? it's just seems wrong.

File: pandas/core/reshape/tile.py [hunk 2]
```diff
@@ -376,9 +376,8 @@ def _bins_to_cuts(
     if len(unique_bins) < len(bins) and len(bins) != 2:
         if duplicates == "raise":
             raise ValueError(
-                "Bin edges must be unique: {bins!r}.\nYou "
-                "can drop duplicate edges by setting "
-                "the 'duplicates' kwarg".format(bins=bins)
+                f"Bin edges must be unique: {repr(bins)}.\n "
```
Comment (MomIsBestFriend) [inline pandas/core/reshape/tile.py]: Should I replace the ```\n``` with a ```;```?

File: pandas/io/formats/style.py [hunk 3]
```diff
@@ -627,29 +627,24 @@ def _apply(self, func, axis=0, subset=None, **kwargs):
...
         if result_shape != expected_shape:
-            msg = (
-                "Function {func!r} returned the wrong shape.\n"
-                "Result has shape: {res}\n"
-                "Expected shape:   {expect}".format(
-                    func=func, res=result.shape, expect=expected_shape
-                )
+            raise ValueError(
+                f"Function {repr(func)} returned the wrong shape.\n"
+                f"Result has shape: {result.shape}\n"
```
Comment (MomIsBestFriend) [inline pandas/io/formats/style.py]: Same here.

**Comment by jreback** [comment]
cc @simonjayhawkins @WillAyd

**Review by WillAyd — APPROVED** [review]
lgtm if you care to look @simonjayhawkins

File: pandas/core/reshape/tile.py (hunk 2 above)
Comment (simonjayhawkins) [reply pandas/core/reshape/tile.py]: is there an extra space here now before the You?

**Review by simonjayhawkins — COMMENTED** [review]
@MomIsBestFriend lgtm. one question.

File: pandas/core/reshape/tile.py (hunk 2 above)
Comment (MomIsBestFriend) [reply pandas/core/reshape/tile.py]: @simonjayhawkins Yes, you're absolutely right.

File: pandas/core/nanops.py (hunk 1 above)
Comment (jreback) [reply pandas/core/nanops.py]: its fine

File: pandas/core/nanops.py [hunk 4]
```diff
@@ -60,8 +60,10 @@ def __call__(self, f):
...
+                raise TypeError(
+                    f"reduction operation {repr(f_name)} not allowed for this dtype"
```
Comment (jreback) [inline pandas/core/nanops.py]: remove the repr

File: pandas/core/reshape/concat.py [hunk 5]
```diff
@@ -542,8 +542,8 @@ def _get_concat_axis(self) -> Index:
                 for i, x in enumerate(self.objs):
                     if not isinstance(x, Series):
                         raise TypeError(
-                            "Cannot concatenate type 'Series' "
-                            "with object of type {type!r}".format(type=type(x).__name__)
+                            f"Cannot concatenate type 'Series' with object "
+                            f"of type {repr(type(x).__name__)}"
```
Comment (jreback) [inline pandas/core/reshape/concat.py]: same

File: pandas/core/sorting.py [hunk 6]
```diff
@@ -208,7 +208,7 @@ def lexsort_indexer(keys, orders=None, na_position="last"):
             cat = Categorical(key, ordered=True)
 
         if na_position not in ["last", "first"]:
-            raise ValueError("invalid na_position: {!r}".format(na_position))
+            raise ValueError(f"invalid na_position: {repr(na_position)}")
```
Comment (jreback) [inline pandas/core/sorting.py]: remove this repr

File: pandas/core/strings.py [hunk 7]
```diff
@@ -1932,13 +1932,10 @@ def _forbid_nonstring_types(func):
         @wraps(func)
         def wrapper(self, *args, **kwargs):
             if self._inferred_dtype not in allowed_types:
-                msg = (
-                    "Cannot use .str.{name} with values of inferred dtype "
-                    "{inf_type!r}.".format(
-                        name=func_name, inf_type=self._inferred_dtype
-                    )
+                raise TypeError(
```
Comment (jreback) [inline pandas/core/strings.py]: same

**Review by jreback — CHANGES_REQUESTED** [review]
most of the repr can be removed here (you can leave the ones for more complex objects, e.g. bins

File: pandas/core/missing.py [hunk 8]
```diff
@@ -211,9 +211,10 @@ def interpolate_1d(
     valid_limit_directions = ["forward", "backward", "both"]
     limit_direction = limit_direction.lower()
     if limit_direction not in valid_limit_directions:
-        msg = "Invalid limit_direction: expecting one of {valid!r}, got {invalid!r}."
         raise ValueError(
-            msg.format(valid=valid_limit_directions, invalid=limit_direction)
+            f"Invalid limit_direction: expecting one of "
+            f"{repr(valid_limit_directions)}, "
```
Comment (jreback) [inline pandas/core/missing.py]: the reprs are not needed here (these are strings).

**Review by jreback — CHANGES_REQUESTED** [review]
one repr to remove, ping on green.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm ping on green

**Comment by MomIsBestFriend** [comment]
ping @jreback

**Comment by simonjayhawkins** [comment]
Thanks @MomIsBestFriend

**Pull request merged**